caml_z.c

```diff
@@ -398,7 +398,7 @@ static value ml_z_reduce(value r, mp_size_t sz, intnat sign)
 #if Z_USE_NATINT
   if (!sz) return Val_long(0);
   if (sz <= 1 && Z_LIMB(r)[0] <= Z_MAX_INT) {
-    if (sign) return Val_long(-Z_LIMB(r)[0]); 
+    if (sign) return Val_long(0-Z_LIMB(r)[0]);
```

Comment:
The visual studio compiler is pretty strict and has  a lot of warnings in contrast to what it supports. Just adding a `0` before the unary minus removes the warning. Another possibility would be to just disable the warning completely.